econml/sklearn_extensions/model_selection.py

```diff
@@ -255,7 +260,112 @@ def get_n_splits(self, X, y, groups=None):
+        self.n_jobs = n_jobs
+        self.refit = refit
+        self.grid_folds = grid_folds
+        self.verbose = verbose
+        self.random_state = random_state
+        self.pre_dispatch = pre_dispatch
+        self.error_score = error_score
+        self.return_train_score = return_train_score
+        self.is_discrete = is_discrete
 
+    def fit(self, X, y, *, sample_weight=None, groups=None):
+        """
+        Perform cross-validation on the estimator list.
+        """
+        self._search_list = []
+        
+        if self.scaling:
+            # pdb.set_trace()
+            if is_data_scaled(X):
+                warnings.warn("Data may already be scaled. Scaling twice may negatively affect results.", UserWarning)
+            self.scaler = StandardScaler()
+            self.scaler.fit(X)
+            scaled_X = self.scaler.transform(X)
+
+        for estimator, param_grid in zip(self.complete_estimator_list, self.param_grid_list):
+            try:
+                if self.random_state != None:
+                    # pdb.set_trace()
+                    if has_random_state(model=estimator):
+                        if is_polynomial_pipeline(estimator):
+                            estimator = estimator.set_params(linear__random_state=42)
+                        else:
+                            estimator.set_params(random_state=42)
+                print(estimator)
+                print(param_grid)
+                temp_search = GridSearchCV(estimator, param_grid, scoring=self.scoring,
+                                       n_jobs=self.n_jobs, refit=self.refit, cv=self.grid_folds, verbose=self.verbose,
+                                       pre_dispatch=self.pre_dispatch, error_score=self.error_score,
+                                       return_train_score=self.return_train_score)
+                if self.scaling: # is_linear_model(estimator) and
+                    if is_polynomial_pipeline(estimator=estimator):
+                        temp_search.fit(scaled_X, y, groups=groups, linear__sample_weight=sample_weight)
+                    elif is_mlp(estimator=estimator):
+                        temp_search.fit(scaled_X, y, groups=groups)
+                    else:
+                        temp_search.fit(scaled_X, y, groups=groups, sample_weight=sample_weight) # , groups=groups, sample_weight=sample_weight
+                    self._search_list.append(temp_search)
+                else:
+                    if is_polynomial_pipeline(estimator=estimator):
+                        temp_search.fit(X, y, groups=groups, linear__sample_weight=sample_weight)
+                    elif is_mlp(estimator=estimator):
+                        temp_search.fit(X, y, groups=groups)
+                    else:
+                        temp_search.fit(X, y,  groups=groups, sample_weight=sample_weight)
+                    self._search_list.append(temp_search)
+            except (ValueError, TypeError, FitFailedWarning) as e:
```

Comment:
Why only these three errors?  Won't any other errors that are generated cause the same issues?